### ISSUE TITLE
Repeat "~" command failing on aggregated scala.js projects

### ISSUE THREAD

**Issue by Rich2** [sbt/sbt#3499]
(See the guidelines for contributing, linked above)

## steps
1 Create Folder called "sbtFolder" I'm using absolute path names as that's what I use in my real work sbt project. So when reproducing the bug the absolute path names will have to be adjusted.

2 set sbt.version=1.0.1
3 addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.20") 

4 create  sub folder "srcCommon" and place the following scala source in it:

```
object C1
{
	val x = 12
} 
```

5 create sub folder "srcA and place the following scala source in it: 
```
object AppA extends scala.scalajs.js.JSApp
{
   def main(): Unit =
   {
   println("This is A " + C1.x.toString) 
   }
}   
```

6 create sub folder "srcB"  and place the following scala source in it:
```
object AppA extends scala.scalajs.js.JSApp
{
   def main(): Unit =
   {
   println("This is B" + (C1.x + 5).toString) 
   } 
} 
```
7 create the following build.sbt file:
```
scalaVersion in ThisBuild := "2.12.3"

lazy val common = Project("common", file("common")).enablePlugins(ScalaJSPlugin).settings(
   scalaSource in Compile := file("/sdat/sbtFolder/srcCommon"))
lazy val projA = Project("proja", file("proja")).enablePlugins(ScalaJSPlugin).dependsOn(common).settings(
   scalaSource in Compile := file("/sdat/sbtFolder/srcA"))
lazy val projB = Project("projb", file("projb")).enablePlugins(ScalaJSPlugin).dependsOn(common).settings(
   scalaSource in Compile := file("/sdat/sbtFolder/srcB"))

lazy val agg = project.aggregate(projA, projB)
```

8 run sbt
9 project agg
10  ~ fastOptJS
11 modify C1.scala and save

## problem
the projects are not recompiled and fastOptJs is not re-run. This worked for many many months in the sbt 0.13 series in my own real work project, but I have not tested this test project in sbt 0.13. When I run proja or projb and then call "~ fastOptJS" the repeat command works as expected recompiling on source changes to the C1 object.

**Comment by Duhemm** [comment]
@Richtype Thanks for the report. I can't reproduce the behaviour you expect on 0.13.15. Can you try again and make sure that it worked the way you describe in 0.13.15?

**Comment by Rich2** [comment]
Apologies the statement about 0.13 was not entirely accurate. I was running repeat "~" Scala.js fastOptJS and fullOptJS commands on an aggregated project (6 or 7 scala.js sub projects) hours a day, nearly every day for many months though multiple versions of sbt 0.13 and multiple versions of scala.js. So the functionality definitely worked for me on sbt 0.13.

When I switched to sbt 1.0.0 I discovered the functionality no longer worked. I tried to return to sbt 0.13, but despite a couple of attempts I wasn't able to get sbt 0.13 to work again on my project. I will edit the opening comment.
